File: javascript/classes/params.js
class Fraction {
  constructor(numerator, denominator = 1) {
    this.numerator = numerator;
    this.denominator = denominator;
  }

  getNumerator() {
    return this.numerator;
  }

  getDenominator() {
    return this.denominator;
  }

  getValue() {
    console.log(`${this.getNumerator()} / ${this.getDenominator()}`);
  }
}

const fraction = new Fraction(5, 10);
fraction.getValue();

File: c++/classes/params.cpp
#include <iostream>
#include <cassert>

class Fraction
{
private:
  int m_numerator;
  int m_denominator;

public:
  Fraction() // default constructor
  {
    m_numerator = 0;
    m_denominator = 1;
  }

  Fraction(int numerator, int denominator = 1)
  {
    assert(denominator != 0);
    m_numerator = numerator;
    m_denominator = denominator;
  }

  int getNumerator() { return m_numerator; }
  int getDenominator() { return m_denominator; }
  double getValue() { return static_cast<double>(m_numerator) / m_denominator; }
};

int main()
{
  Fraction frac;
  std::cout << frac.getNumerator() << "/" << frac.getDenominator() << '\n';

  return 0;
}
